File: files/2025-09-28/CSEC 28092025.R
library(tidyverse)
library(readxl)

path = "files/2025-09-28/Challenge 67.xlsx"
input = read_excel(path, range = "B2:E8")
test  = read_excel(path, range = "G2:J7")

result = input %>%
  mutate(Prices = Prices + first(Prices),
         Values = `Unit Values` + first(`Unit Values`)) %>%
  filter(row_number() != 1) %>%
  select(-`Unit Values`) 

all.equal(result, test)
# [1] TRUE

File: files/2025-09-28/CSEC 28092025.py
import pandas as pd

path = "files/2025-09-28/Challenge 67.xlsx"
input = pd.read_excel(path, usecols="B:E", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="G:J", skiprows=1, nrows=5).rename(columns=lambda c: c.replace('.1', ''))

numeric_cols = input.select_dtypes(include='number').columns
for col in numeric_cols:
    input[col] = input[col] + input[col].iloc[0]
result = input.iloc[1:].reset_index(drop=True)

result.columns = test.columns
print(result.equals(test)) # True